documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,213 @@
++------------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a 32-bit or 64-bit bit C++ integral or floating-point type.
+- The result is non-deterministic if the binary operator is non-associative (such as in floating-point addition)
+  or non-commutative.
+
+
+.. note::
+
+  Current limitations:
+
+  - The function will internally block until the issued kernels have completed execution.
+    Although intended in the future to be an asynchronous call, the algorithm is currently synchronous.
+  - The SYCL device associated with the provided queue must support 64-bit atomic operations if the element type is 64-bits.
+  - The binary operator must have a known identity. That is, ``sycl::has_known_identity<Op>::value`` must
+    evaluate to true. Such operators are listed in the `SYCL 2020 specification <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.identities>`_.
+
+Return Value
+------------
+
+A ``sycl::event`` object representing the status of the algorithm execution.
+
+--------------
+Usage Examples
+--------------
+
+
+inclusive_scan Example
+----------------------
+
+.. code:: cpp
+
+   // possible build and run commands:
+   //    icpx -fsycl inclusive_scan.cpp -o inclusive_scan -I /path/to/oneDPL/include && ./inclusive_scan
+
+   #include <cstdint>
+   #include <iostream>
+   #include <sycl/sycl.hpp>
+
+   #include <oneapi/dpl/experimental/kernel_templates>
+
+   namespace kt = oneapi::dpl::experimental::kt;
+
+   int main()
+   {
+      std::size_t n = 6;
+      sycl::queue q{sycl::gpu_selector_v};
+      std::uint32_t* arr = sycl::malloc_shared<std::uint32_t>(n, q);
+      std::uint32_t* out = sycl::malloc_shared<std::uint32_t>(n, q);
+
+      // initialize
+      arr[0] = 1, arr[1] = 2, arr[2] = 1, arr[3] = 3, arr[4] = 1, arr[5] = 2;
+
+      // scan
+      auto e = kt::gpu::inclusive_scan(q, arr, arr + n, out, std::plus<std::uint32_t>{}, kt::kernel_param<256, 8>{});
+      e.wait();
+
+      // print
+      for(std::size_t i = 0; i < n; ++i)
+         std::cout << out[i] << ' ';
+      std::cout << '\n';
+
+      sycl::free(arr, q);
+      sycl::free(out, q);
+      return 0;
+   }
+
+**Output:**
+
+.. code:: none
+
+   1 3 4 7 8 10
+
+.. _memory-requirements:
+
+-------------------
+Memory Requirements
+-------------------
+
+The algorithms use global and local device memory (see `SYCL 2020 Specification
+<https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_device_memory_model>`_)
+for intermediate data storage. For the algorithms to operate correctly, there must be enough memory
+on the device; otherwise, the algorithm throws a ``std::bad_alloc`` exception. The amount of memory that is required
+depends on input data and configuration parameters, as described below.
+
+Global Memory Requirements
```

Comment:
I suppose there is no need in a such elaborate excursion into the global memory requirements as we have ``bad_alloc`` for the user as a tool to handle the corresponding issue.

What about limiting the description of the whole section to: 
> The amount of the internally allocated memory does not exceed the amount needed to store the input sequence and usually it is a small fracture of it.

Reply:
I believe there is still some value in informing the user about how much global memory is required. I agree that throwing `bad_alloc` will tell the user that their current configuration cannot be satisfied, but it will probably still be helpful to let users know what type of global memory usage to expect and how that usage depends on their kernel parameters.

Reply:
I still believe that it is too detailed that may confuse a user. One would need to understand that section before starting using that functionality and it is not simple.

What do you think about removing "upper bound" and "adding 33 for padding" parts to reduce verbosity?

Reply:
Or alternatively, accompany the verbose estimation with a rough one.

Reply:
I agree there was a lot of verbosity there. I removed a lot of the unnecessary details such as the padding and the variable for the flag length.

Reply:
It looks much simpler now, thanks